tests/chainer_tests/test_link.py

```diff
@@ -1194,10 +1212,17 @@ def test_copy_with_init_mode(self):
         self.l2.x.initializer = initializers.Normal(
             dtype=self.l2.x.initializer.dtype)
         self.l2.x.initialize(self.l2.x.shape)
+        self.x.initializer = initializers.Normal(
+            dtype=self.x.initializer.dtype)
+        self.c2.x.initialize(self.x.shape)
 
         c2 = self.c2.copy(mode='init')
         self.assertIs(c2.name, None)
         self.assertIsInstance(c2._children, set)
+        self.assertTrue(hasattr(c2, 'x'))
+        self.assertIsNot(c2.x, self.x)
+        self.assertIsNot(c2.x.data, self.x.data)
```

Comment:
How about this check: `self.assertFalse(numpy.array_equal(c2.x.data, self.x.data))`